osu.Game/Screens/OnlinePlay/Multiplayer/MultiplayerMatchSongSelect.cs

```diff
@@ -76,40 +72,23 @@ protected override void SelectItem(PlaylistItem item)
 
                 Task task = itemToEdit != null ? client.EditPlaylistItem(multiplayerItem) : client.AddPlaylistItem(multiplayerItem);
 
-                task.ContinueWith(t =>
+                task.FireAndForget(onSuccess: () => Schedule(() =>
                 {
-                    Schedule(() =>
-                    {
-                        // If an error or server side trigger occurred this screen may have already exited by external means.
-                        if (!this.IsCurrentScreen())
-                            return;
+                    // If an error or server side trigger occurred this screen may have already exited by external means.
+                    if (!this.IsCurrentScreen())
+                        return;
 
-                        loadingLayer.Hide();
-
-                        if (t.IsFaulted)
-                        {
-                            Exception exception = t.Exception;
-
-                            if (exception is AggregateException ae)
-                                exception = ae.InnerException;
-
-                            Debug.Assert(exception != null);
-
-                            string message = exception is HubException
-                                // HubExceptions arrive with additional message context added, but we want to display the human readable message:
-                                // "An unexpected error occurred invoking 'AddPlaylistItem' on the server.InvalidStateException: Can't enqueue more than 3 items at once."
-                                // We generally use the message field for a user-parseable error (eventually to be replaced), so drop the first part for now.
-                                ? exception.Message.Substring(exception.Message.IndexOf(':') + 1).Trim()
-                                : exception.Message;
-
-                            Logger.Log(message, level: LogLevel.Important);
-                            Carousel.AllowSelection = true;
-                            return;
-                        }
+                    loadingLayer.Hide();
+                    this.Exit();
```

Comment:
Given that the `onSuccess` and `onError` paths here are quite similar, maybe worth splitting a helper? Not sure.

Reply:
A local method local to this class you mean? Something like

```csharp
FireAndForget(() => handle(null), handle)

handle(Exception e)
{
...
}
```

?

Reply:
Yeah, just something local to remove the duplication of the current screen check and hiding the loading layer. Although I'm not entirely 100% it's worth the effort.

Reply:
Not sure it's worth it. I've refactored the code a bit, as the `IsCurrentScreen` check wasn't required for most of it and was just a safety.